06-building-a-program-start-to-finish/solving-the-problem/output/invertedindex.js

```diff
@@ -0,0 +1,222 @@
+          indexWords[token].push(eachDoc.documentNum);
+        }
+      });
+    });
+    return indexWords;
+  }
+
+  /**
+   * getIndex method returns the indexed words and the documents that were found
+   * @param filename {string} : name of the file to get its index
+   * @return {Object} or {boolean} the words index
+  */
+  getIndex(filename) {
+    try {
+      if (!this.filesIndexed[filename]) {
+        this.error.status = false;
+        this.error.message = 'File selected not indexed';
+        throw this.error;
+      }
+      const file = this.filesIndexed[filename];
+      return file.index;
+    } catch (err) {
+      return this.error.status;
+    }
+  }
+
+  /**
+   * searchIndex searches the indexed words to determine the
+   * documents that the searchterms can be found
+   * @param {string, array} searchTerm - the search query
+   * @param {string} filename - the name of the file to search its index
+   * @return {object|boolean} it returns boolean if the searchTerm is empty and
+   * it returns object if it is not. Each index is each searcykeyword.
+   * Each with an array value of the document index
+  */
+  searchIndex(searchTerm, filename) {
+    if ((typeof searchTerm === 'string' && searchTerm.trim() === '') ||
+      (typeof searchTerm === 'object' && searchTerm.length === 0) ||
+      searchTerm === undefined) {
+      return false;
+    }
+    const result = [];
+    if (filename === 'all') {
+      Object.keys(this.filesIndexed).forEach((eachFile) => {
+        result.push({
+          indexes: this.getSearchResults(searchTerm, eachFile),
+          searchedFile: eachFile,
+          documents: this.getDocuments(eachFile)
+        });
+      });
+    } else {
+      result.push({
+        indexes: this.getSearchResults(searchTerm, filename),
+        searchedFile: filename,
+        documents: this.getDocuments(filename)
+      });
+    }
+    return result;
+  }
+
+  /**
+   * getSearchResults method checks the index of the file and returns the result
+   * @param searchTokens {searchTokens} - the search query of one or more words
+   * @param filename {string} - the name of the file
+   * @return {object} - an object with the found words as keys
+  */
+  getSearchResults(searchTokens, filename) {
+    const indexToSearch = this.getIndex(filename), result = {};
+    this.tokenize(searchTokens).forEach((eachSearchWord) => {
+      if (indexToSearch[eachSearchWord]) {
+        result[eachSearchWord] = indexToSearch[eachSearchWord];
+      }
+    });
+    return result;
+  }
+
+  /**
+   * getDocuments get an array of the documents index e.g [0, 1, 2, 3]
+   * @param {filename} - name of the file to get its document
+   * @return {array} an array of the documents index
+  */
+  getDocuments(filename) {
+    const docs = [];
+    for (let i = 0; i < this.filesIndexed[filename].numOfDocs; i += 1) {
+      docs.push(i);
+    }
+    return docs;
+  }
+
+}
```

Comment:
- Your JSDoc comments in this file are not correctly done.
- JSDoc syntax errors
- You have several lines exceeding the maximum line length of 80
- _**Expected 'this' to be used by class method**_ on almost all your methods.
- Improper/abbreviated variable/parameter names

Please fix these issues